src/middleware.ts

```diff
@@ -26,11 +26,18 @@ async function IPXReqHandler (req: IncomingMessage, res: ServerResponse, ipx: IP
     throw badRequest('Missing src')
   }
 
+  let [fileFormat, responseFormat] = format === '_' ? ['_', 'image'] : format.split('_')
+  if (fileFormat === 'json') {
+    fileFormat = '_'
+    responseFormat = 'json'
+  }
   // Get basic info about request
-  const info = await ipx.getInfo({ adapter, format, operations, src })
+  const info = await ipx.getInfo({ adapter, format: fileFormat, operations, src })
 
   // Set Content-Type header
-  if (info.format) {
+  if (responseFormat === 'json') {
+    res.setHeader('Content-Type', 'application/json')
```

Comment:
We may generate `contentType` inside `getInfo` to avoid duplicate logic here to detect JSON